src/doc/rustc/src/codegen-options/index.md

```diff
@@ -367,12 +401,35 @@ machine. Each target has a default base CPU.
 
 ## target-feature
 
-Individual targets will support different features; this flag lets you control
-enabling or disabling a feature. Each feature should be prefixed with a `+` to
-enable it or `-` to disable it. Separate multiple features with commas.
+This option controls either CPU instruction set extensions that can be used during code generation,
+or static linking.
+
+In both cases `-C target-feature=+x` enables feature `x` and `-C target-feature=-x` disables it.
+
+Multiple features can be specified by separating them with commas - `-C target-feature=+x,-y`.
 
-To see the valid options and an example of use, run `rustc --print
-target-features`.
+For each supported feature `x` `target_feature = "x"` becomes a `cfg` predicate usable early during
+compilation.  
+Code configured with `cfg(target_feature = "x")` is kept if the feature `x` is enabled,
+and removed otherwise.
+
+If the enabled feature is not supported for the current target, the behavior is unspecified.
+<!-- There are multiple bugs here
+  - Unknown CPU features go straight to LLVM and appear as LLVM warnings.
+  - At configure time `crt-static` predicate is set even if the target doesn't support `crt-static`.
+  - At link time `crt-static` is sometimes ignored
+    if the target doesn't support it and sometimes not.
+-->
+
+To see the valid features and an example of use, run `rustc --print target-features`.
+
+#### CPU instruction set extensions
+
+Target features that allow enabling or disabling architecture-dependent CPU features,
+like `+neon` or `-sse`.
```

Comment:
This sentence seems a little strange without a verb.  Maybe something like this?

> Target features for CPU instruction set extensions can enable or disable architecture-dependent CPU features, like `+neon` or `-sse`.